DataManagementSystem/Client/ConsistencyChecks.py

```diff
@@ -0,0 +1,836 @@
+    if failedLfns:
+      return S_ERROR( 'Failed to obtain replicas' )
+
+
+    # For the LFNs found to be registered obtain the file metadata from the catalog and verify against the storage metadata
+    res = self.__getCatalogMetadata( storageMetadata )
+    if not res['OK']:
+      return res
+    catalogMetadata = res['Value']
+    sizeMismatch = []
+    for lfn, lfnCatalogMetadata in catalogMetadata.items():
+      lfnStorageMetadata = storageMetadata[lfn]
+      if ( lfnStorageMetadata['Size'] != lfnCatalogMetadata['Size'] ) and ( lfnStorageMetadata['Size'] != 0 ):
+        sizeMismatch.append( ( lfn, 'deprecatedUrl', storageElement, 'CatalogPFNSizeMismatch' ) )
+    if sizeMismatch:
+      self.__reportProblematicReplicas( sizeMismatch, storageElement, 'CatalogPFNSizeMismatch' )
+    gLogger.info( 'Checking storage files exist in the catalog complete' )
+    resDict = {'CatalogMetadata':catalogMetadata, 'StorageMetadata':storageMetadata}
+    return S_OK( resDict )
+
+  def getStorageDirectoryContents( self, lfnDir, storageElement ):
+    """ This obtains takes the supplied lfn directories and recursively obtains the files in the supplied storage element
+    """
+    return self.__getStorageDirectoryContents( lfnDir, storageElement )
+
+  def __getStorageDirectoryContents( self, lfnDir, storageElement ):
+    """ Obtians the contents of the supplied directory on the storage
+    """
+    gLogger.info( 'Obtaining the contents for %s directories at %s' % ( len( lfnDir ), storageElement ) )
+
+    se = StorageElement( storageElement )
+
+    res = se.exists( lfnDir )
+    if not res['OK']:
+      gLogger.error( "Failed to obtain existance of directories", res['Message'] )
+      return res
+    for directory, error in res['Value']['Failed'].items():
+      gLogger.error( 'Failed to determine existance of directory', '%s %s' % ( directory, error ) )
+    if res['Value']['Failed']:
+      return S_ERROR( 'Failed to determine existance of directory' )
+    directoryExists = res['Value']['Successful']
+    activeDirs = []
+    for directory in sorted( directoryExists ):
+      exists = directoryExists[directory]
+      if exists:
+        activeDirs.append( directory )
+    allFiles = {}
+    while len( activeDirs ) > 0:
+      currentDir = activeDirs[0]
+      res = se.listDirectory( currentDir )
+      activeDirs.remove( currentDir )
+      if not res['OK']:
+        gLogger.error( 'Failed to get directory contents', res['Message'] )
+        return res
+      elif currentDir in res['Value']['Failed']:
+        gLogger.error( 'Failed to get directory contents', '%s %s' % ( currentDir, res['Value']['Failed'][currentDir] ) )
+        return S_ERROR( res['Value']['Failed'][currentDir] )
+      else:
+        dirContents = res['Value']['Successful'][currentDir]
+        activeDirs.extend( se.getLFNFromURL( dirContents['SubDirs'] ).get( 'Value', {} ).get( 'Successful', [] ) )
```

Comment:
SubDirs and Files contains LFN, no URLs so the calls to getLFNFromURL are useless